SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1441 @@
+						$eachAttribute.Value = "NULL"
+						$eachNode.productAttributes.driftAlarm = "Red"
+						$count += 1
+					}
+				}
+			}
+		}
+	}
+	#Getting vcenter info
+	$vcenter = Get-vCenterServerDetail -server $ppmVariables.sddcManagerFqdn -user $ppmVariables.sddcManagerUser -pass $ppmVariables.sddcManagerPass -domain $ppmVariables.sddcDomainName
+	if (!$ppmVariables.esxiCluster) {
+		$clusterID = Get-VCFWorkloadDomain -name $ppmVariables.sddcDomainName
+		$ppmVariables.esxiCluster = $clusterID
+	}
+
+	#start setting password policy based on driftAlarm
+	#VC
+	if ($ppmEnvironmentalDetails | where { ($_.productType -match "VC") -and ($_.productAttributes.driftAlarm -match "Red") } ) {
+		$products = $ppmStandardConfigValues | Select-Object -Skip 0 | Where-Object -FilterScript { $_.productType -eq "VC" }
+		$passwordTtlInDays = $products.productAttributes.passwdExpInDays
+		$emailNotification = $products.productAttributes.passwdNotifyEmail
+		Write-LogMessage -type INFO -Message "Setting Password Policy on vCenter Server Appliance.  Starting..."
+		Set-vCenterPasswordExpiration -server $ppmVariables.sddcManagerFqdn -user $ppmVariables.sddcManagerUser -pass $ppmVariables.sddcManagerPass -domain $ppmVariables.sddcDomainName -passwordExpires $true -email $emailNotification -maxDaysBetweenPasswordChange $passwordTtlInDays | Out-Null
+		Write-LogMessage -type INFO -Message "Setting Password Policy on vCenter Server Appliance.  Complete."
+	}
+	
+	#SSO
+	if ($ppmEnvironmentalDetails | where { ($_.productType -match "SSO") -and ($_.productAttributes.driftAlarm -match "Red") } ) {
+		$products = $ppmStandardConfigValues | Select-Object -Skip 0 | Where-Object -FilterScript {$_.productType -eq "SSO"}
+		$passHistory = $products.productAttributes.passwdHistoryRestriction
+		$minCharacterLength = $products.productAttributes.passwdMinimumLength
+		$maxCharacterLength = $products.productAttributes.passwdMaximumLength
+		$minNumericChar = $products.productAttributes.passwdMinNumeric
+		$minSpecialChar = $products.productAttributes.passwdMinSpecial
+		$minUppercaseChar = $products.productAttributes.passwdMinUppercase
+		$minLowercaseChar = $products.productAttributes.passwdMinLowercase
+		$maxConsecutiveIdenticalChar = $products.productAttributes.passwdMaxConsecutiveIdenticalChar
+		if ([int]$products.productAttributes.passwdMinAlphabetic -lt [int]([int]$minUppercaseChar + [int]$minLowercaseChar)) {
+			$minAlphabeticCount = [int]$minUppercaseChar + [int]$minLowercaseChar
+		} else {
+			$minAlphabeticCount = $products.productAttributes.passwdMinAlphabetic
+		}
+		$passwordTtlInDays = $products.productAttributes.passwdExpInDays
+		$autoUnlockIntervalSec = $products.productAttributes.passwdUnlockIntervalInSec
+		$failedAttemptIntervalSec = $products.productAttributes.passwdAttemptsIntervalInSec
+		$numAttempts = $products.productAttributes.passwdMaxFailAttempts
+		Write-LogMessage -type INFO -Message "Setting Password Policy on vCenter SSO services.  Starting..."
```

Comment:
Change From: 'SSO services'
To: 'Single-Sign On Domain'

Reply:
addressed in 99029d5ec8015e0c5c645e9abc7c3384e1b3091e